src/stream/binary_writer.rs

```diff
@@ -527,7 +526,7 @@ impl<W: Write> BinaryWriter<W> {
                     self.writer.write_exact(&buf)?;
                 } else {
                     let mut buf: [_; 9] = [0x87, 0, 0, 0, 0, 0, 0, 0, 0];
-                    buf[1..].copy_from_slice(&(v as u64).to_be_bytes());
+                    buf[1..].copy_from_slice(&v.to_be_bytes());
```

Comment:
Please keep this, or change to `u64::to_be_bytes(v)`. I want it to be very clear what the type of `v` is.

Reply:
Part of the second commit, so this will be dropped out of the PR.